x-pack/legacy/plugins/ingest/server/libs/adapters/framework/default.ts

```diff
@@ -52,12 +56,24 @@ export class BackendFrameworkAdapter {
     }
   }
 
+  public async waitForStack() {
+    return new Promise(resolve => {
+      this.on('xpack.status.green', () => {
+        resolve();
+      });
+    });
+  }
+
   public getSetting(settingPath: string) {
     return this.server.config().get(settingPath);
   }
 
   public log(text: string) {
-    this.server.log(text);
+    if (this.server) {
```

Comment:
why server could be undefined here?

Reply:
Your guess is as good as mine. I just know sometimes I get an error that it is